dora/core/common/src/main/java/alluxio/wire/WorkerNetAddress.java

```diff
@@ -233,6 +254,8 @@ public boolean equals(Object o) {
         && mDataPort == that.mDataPort
         && mWebPort == that.mWebPort
         && mDomainSocketPath.equals(that.mDomainSocketPath);
+        // Skip the comparison of mHttpServerPort for backward compatibility
+        // && mHttpServerPort == that.mHttpServerPort;
```

Comment:
what's incompatible with comparing the restful api port for equality?

Reply:
also update the `hashCode` implementation, or leave a comment explaining why `mHttpServerPort` is excluded from computing the hashcode.

Reply:
> Incompatibility.
As @lucyge2022  mentioned, the previous etcd-registered worker information doesn't have HTTP server ports. 1) When we do the comparison, the lack of HTTP ports will cause an issue. 2) It will also influence worker registration if we just add an HTTP port for an old worker (I added a new test to cover this scenario).

Reply:
> hashCode.

Updated with a comment on the exclusion of `mHttpServerPort`.

Reply:
> When we do the comparison, the lack of HTTP ports will cause an issue

When checking equality, `WorkerServiceEntity` explicitly ignores all ephemeral information about a worker such as network addresses, and only compares the worker identity. This is why `WorkerIdentity` exists. Therefore, a different HTTP port won't cause issue, as any other existing network addresses like hostname and grpc port.